### PR TITLE
Proposal: add ingress exposure as part of the routing rule

### PR DESCRIPTION
**Pull request by kyessenov**
follow up from @rshriram, @ijsnellf discussion

### CODE REVIEW

File: proxy/v1/config/route_rule.proto [hunk 1]
```diff
@@ -134,6 +134,28 @@ message RouteRule {
...
+}
+
+// Ingress specification for exposing a destination port externally
+message Ingress {
+  // REQUIRED: Port to expose the destination service on.
+  int32 port = 1;
+  
+  // Optional TLS secret path to apply on the exposed port.
+  string tls_secret = 2;
+
```
Comment (rshriram) [inline proxy/v1/config/route_rule.proto]: might want to add client_secret and server_secret as well, to support mTLS on ingress.

**Comment by frankbu** [comment]
@rshriram does the ingress rule have a match (patch/host) clause in your proposal, or do we still need to create a rule to block unwanted paths and rules to allow selected paths?

**Comment by kyessenov** [comment]
The port mapping is important for cloud provider LBs. They typically don't allow exposing arbitrary ports on the cluster. Without port translation in route rules, it would be impossible to expose a non-default port.

**Comment by rshriram** [comment]
Yes, but my point is that this port mapping does not have to be part of the route rule spec.. i.e., if service wants to listen on port 80 because the provider allows port 80, then all is fine. If service insists on port 9080, then the user has to find another way of mapping.

Alternatively, you can imagine specifying something like this:

```yaml
tlsMappings:
- host:
    name: foo.bar.com
    serverSecret: baz
    clientSecret: bar
...
applyRules:
- port: 80
   protocol: http (includes gRPC/H2)
   ruleLabels:
      foo: bar
- port: 443
   protocol: https
  ruleLabels:
     foo: baz
- port: 9379
   protocol: tcp
   service: myTCPservice
```

**Comment by rshriram** [comment]
The above is no different from the HTTP proxy configuration

**Comment by rshriram** [comment]
closing in favor of #254 and #243

**Pull request closed without merging**